scripts/check-headers-within-dir.sh

```diff
@@ -27,21 +27,23 @@ rm -f "$PWD"/.current_*.cc "$PWD"/.current_*.o "$PWD"/.current_.txt "$SOURCE_FIL
 
 echo -e -n "\033[1mCompiling\033[0m: "
 for i in $(ls *.h | grep -v ".cc.h$" | grep -v "^current_build.h") ; do
-  echo -e -n "\033[36m"
-  echo -n "$i "
-  echo -e -n "\033[31m"
-  HEADER_GCC="$PWD/.current_$i.g++.cc"
-  HEADER_CLANG="$PWD/.current_$i.clang++.cc"
-  COMBINED_OBJECT="$PWD/.current_${i}_combined"
-
-  ln -sf "$PWD/$i" "$HEADER_GCC"
-  ln -sf "$PWD/$i" "$HEADER_CLANG"
-
-  g++ -I "$EXTRA_INCLUDE_DIR" $CPPFLAGS -c "$HEADER_GCC" -o "${HEADER_GCC}.o" $LDFLAGS \
-    >"$TMP_STDOUT" 2>"$TMP_STDERR" || (cat "$TMP_STDOUT" "$TMP_STDERR" && exit 1)
-  clang++ -I "$EXTRA_INCLUDE_DIR" $CPPFLAGS -c "$HEADER_CLANG" -o "${HEADER_CLANG}.o" $LDFLAGS \
-    >"$TMP_STDOUT" 2>"$TMP_STDERR" || (cat "$TMP_STDOUT" "$TMP_STDERR" && exit 1)
-  ld -r "${HEADER_GCC}.o" "${HEADER_CLANG}.o" -o "${COMBINED_OBJECT}.o"
+  if [ "$(head -n 1 "$i")" != "// Autogenerated schema inferred from input JSON data." ] ; then
```

Comment:
What do you think about more generic
`if !(head -n 1 "$i" | grep -i "auto" > /dev/null) ; then`
?